app/controllers/stripe-setup/company-number/post.controller.js

```diff
@@ -62,6 +63,8 @@ module.exports = async (req, res, next) => {
       })
       if (isSwitchingCredentials) {
         return res.redirect(303, formatAccountPathsFor(paths.account.switchPSP.index, req.account.external_id))
+      } else if (enabledStripeOnboardingTaskList) {
+        return res.redirect(303, formatAccountPathsFor(paths.account.yourPsp.index, req.account && req.account.external_id, req.account.gateway_account_credentials[0].external_id))
```

Comment:
We should have the credential ID in the path params for the request. 
You should be able to access it from `req.params.credentialId` - this is because it's specified as a path param in the paths file `/your-psp/:credentialId/company-number` with the key "credentialsId".

I think this is better than assuming it will be the only credential id (even though this should always be the case)